src/Core/Case/CaseTree.idr

```diff
@@ -35,13 +36,13 @@ mutual
   ||| Case alternatives. Unlike arbitrary patterns, they can be at most
   ||| one constructor deep.
   public export
-  data CaseAlt : List Name -> Type where
+  data CaseAlt : ScopedList Name -> Type where
        ||| Constructor for a data type; bind the arguments and subterms.
-       ConCase : Name -> (tag : Int) -> (args : List Name) ->
-                 CaseTree (args ++ vars) -> CaseAlt vars
+       ConCase : Name -> (tag : Int) -> (args : ScopedList Name) ->
```

Comment:
It's debatable whether the constructor arguments should be listed right-to-left.
Also they would become the most local variables so the nested case tree would
have a type of the form `CaseTree (vars <>< args)` rather than `args` flowing into
the global scope.

Reply:
Left for now as marked as you said it is a debatable question.